doc/kuksa_requirementes.md

```diff
@@ -0,0 +1,805 @@
+    * I want to receive exactly all the signals’ values I requested without errors.
+
+4. As a client I want to `Subscribe` and be able to apply a filter to receive a sensor value with an interval of x milliseconds only if the value has changed.
+   
+5.  As a client I want to `GetMetadata` static Metadata from signals.
+  - Details:
+    * All sensors, actuators, and attributes values for which a client has permission to.
+    * Client must provide the path, paths or wildcard.
+
+6.  As a client I want to `Get` attributes values once.
+
+7.  As a client I want to `Actuate` a target value of an actuator. This value has to be forwarded to the provider of this actuator if the provider is available, otherwise the target value is lost. 
+
+8.  As a client I want to connect and interact with Databroker either if it contains data or not.
+
+10.  As a client I want to know if there is a provider up and running by getting some sensor or actuator value.
+
+11.  As a client I don’t want to `Subscribe` to target values.
+
+12.  As a client I don’t want to `Subscribe` to attributes values.
+
+13.  As a client I don’t want to `Get` target values.
+
+14.  As a client I don’t want to `Set` sensor values.
+
+15.  As a client I don’t want to `Set` attributes values.
+
+### As Databroker
+1. The data broker shall be able to handle catalogs of signals described by the syntax as defined by the COVESA Vehicle Signal Specification (VSS). This relates to all aspects of the VSS syntax definition, which is also called VSS rule set. This implies that the data broker can handle the signal catalog as defined by the COVESA VSS.
+
+2. Databroker shall support at least those metadata elements as defined by the VSS rule set.
+
+### As Provider
+1. As a provider I want to `Set` the value and current value of one or many sensors and/or actuators with its timestamp to Databroker.
+2. 
+3. A provider shall be able to continuously and efficiently (e.g. via grpc stream) _**update**_ the current value of one or multiple sensors and/or actuators with its timestamp.
+
+4. A provider shall be able to `Subscribe` to the databroker to receive actuation requests.
+
+5. As a provider I want to `Set` once attributes values.
+
+6. As a provider I don’t need to but I would like to `Get` sensor values or current values of actuatros?.
+   
+7. As a provider I don’t want to `Set` target values to Databroker.
+
+# Non-Functional Requirements
+
+## Data availability/persistence according to lifecycle of Client, Databroker and Provider
+1. The data broker either ...
+   * Stores last set values during its own lifecycle,
+   * ? stores values during the system's power cycle (i.e., "persists" values over own restarts, or
+   * store values over system's power cycles.
+
+2. How to "reset" values availability if its provider got inactive (without reseting the value)?
+   * -> Client's job (e.g. using timestamp)?
+   * -> Broker's job (e.g. using timestamp + minimal update cycle)?
```

Comment:
I do not have clear opinion, but am torn towards leaving the decision whether a value is still considered as available to the client. This might depend on the circumstances, the use case and the signal and it is therefore hard to make a general decision even per signal. 

However, it would of course be great if the broker could already decide whether the signal is considered as not available anymore, to relive the client from making this evaluation and avoid sending around the data and timestamps. 
What do you think about allowing the client to optionally add a minimal update cycle to the request and then letting the databroker decide per request whether the signal is returned? This would of course add processing load to the databroker.